rmgweb/database/views.py

```diff
@@ -2607,43 +3334,211 @@ def moleculeSearch(request):
                    })
 
 
-def solvationSearch(request):
+# def solvationSearch(request):
+#     """
+#     Creates webpage form to display solvation data upon choosing a solvent and a solute.
+#     """
+#     from rmgweb.database.forms import SolvationSearchForm
+#     form = SolvationSearchForm()
+#     structure_markup = ''
+#     molecule = Molecule()
+#     if request.method == 'POST':
+#         posted = SolvationSearchForm(request.POST, error_class=DivErrorList)
+#         initial = request.POST.copy()
+#
+#         form = SolvationSearchForm(initial, error_class=DivErrorList)
+#         if posted.is_valid():
+#             adjlist = posted.cleaned_data['adjlist']
+#             if adjlist != '':
+#                 molecule.from_adjacency_list(adjlist)
+#                 structure_markup = getStructureInfo(molecule)
+#                 solute_adjlist = molecule.to_adjacency_list()  # obtain full adjlist, in case hydrogens were non-explicit
+#                 solvent = posted.cleaned_data['solvent']
+#                 solvent_temp = posted.cleaned_data['solvent_temp']
+#                 temp = posted.cleaned_data['temp']
+#                 if solvent == '':
+#                     solvent = 'None'
+#                 if solvent_temp == '':
+#                     solvent_temp = 'None'
+#
+#             if 'solvation' in request.POST:
+#                 return HttpResponseRedirect(reverse('database:solvation-data', kwargs={'solute_adjlist': solute_adjlist, 'solvent': solvent,
+#                                                                                        'solvent_temp': solvent_temp, 'temp': temp}))
+#
+#             if 'reset' in request.POST:
+#                 form = SolvationSearchForm()
+#                 structure_markup = ''
+#                 molecule = Molecule()
+#
+#     return render(request, 'solvationSearch.html', {'structure_markup': structure_markup, 'molecule': molecule, 'form': form})
+
+
+def solvationSearchML(request):
+    """
+    Creates webpage form to display solvation data upon choosing a solvent and a solute using a machine learning model.
+    """
+    from rmgweb.database.forms import SolvationSearchMLForm
+    form = SolvationSearchMLForm()
+
+    if request.method == 'POST':
+        posted = SolvationSearchMLForm(request.POST, error_class=DivErrorList)
+        initial = request.POST.copy()
+
+        form = SolvationSearchMLForm(initial, error_class=DivErrorList)
+        if posted.is_valid():
+            solvent_solute_smiles = posted.cleaned_data['solvent_solute_smiles']
+            calc_dGsolv = posted.cleaned_data['calc_dGsolv']
```

Comment:
Instead of unpacking the dict, you could just directly pass the dictionary values into `get_solvation_data_ML`. As pseudocode, something like:
```
if posted.isvalid():
     calc_dict = posted.cleaned_data
     if 'submit' in request.POST:
          solvation_data_results, additional_info_list = get_solvation_data_ML(calc_dict['solvent_solute_smiles'], calc_dict['calc_dGsolv'] ... etc.
```

Reply:
This is no longer needed because we decided to pass the input arguments instead of the output arguments.